examples/example_piecewise_target_function.py

```diff
@@ -0,0 +1,185 @@
+"""
+Example for evolutionary regression on a piecewise target function with one single genome
+=========================================================================================
+
+Example demonstrating the use of Cartesian genetic programming for
+regression on a piecewise target function.
+Demonstrates the use of if/else branching operator gene.
+"""
+
+# The docopt str is added explicitly to ensure compatibility with
+# sphinx-gallery.
+docopt_str = """
+   Usage:
+     example_piecewise_target_function.py [--max-generations=<N>]
+
+   Options:
+     -h --help
+     --max-generations=<N>  Maximum number of generations [default: 5000]
+"""
+
+import functools
+import matplotlib.pyplot as plt
+import numpy as np
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+
+# %%
+# First we define the new operator node: If/ElseOperator for conditional outputs
+class IfElseOperator(cgp.OperatorNode):
+    """A node that checks its input0 is bigger than 0
+     and outputs input1 if true and input2 otherwise """
+
+    _arity = 3
+
+    def __init__(self, idx, inputs):
+        super().__init__(idx, inputs)
+        if inputs[0] > 0:
+            self._output = inputs[1]
+        else:
+            self._output = inputs[2]
+
+    _def_output = "x_1 if x_0 > 0 else x_2"
```

Comment:
```suggestion
    _def_output = "x_1 if x_0 >= 0 else x_2"
```
this is how i would define it, but if you feel strongly about it, i can live with `x_0 > 0` ;)